### PR TITLE
added Twitter to connect 

### PR DESCRIPTION
**Pull request by Indrajeety993648**
## Description

This pull request introduces support for displaying a Twitter link in the footer component of the application. The following changes have been made:

- **Updated `Footer.tsx` Component:**
  - Added Twitter icon and link to the `Connect` section.
  - Conditionally rendered the Twitter icon based on the presence of `NEXT_PUBLIC_TWITTER_URL` in the environment variables.

- **Updated Environment Configuration (`env.mjs`):**
  - Added `NEXT_PUBLIC_TWITTER_URL` to handle the optional Twitter URL.

- **Fixed Formatting and Style Issues:**
  - Corrected import formatting to comply with Prettier rules.
  - Replaced arbitrary Tailwind CSS values with predefined classes for consistency.

### `Footer.tsx`
- Added Twitter support to the `Connect` section.
- Adjusted import statements for improved readability.

### `env.mjs`
- Included `NEXT_PUBLIC_TWITTER_URL` in the environment configuration.

## Checklist
- [x] Code adheres to the project's style guidelines.
- [x] All new and existing tests pass.
- [x] Changes tested locally.

## Testing Instructions
1. Ensure that your `.env` file includes the `NEXT_PUBLIC_TWITTER_URL` variable.
2. Start the development server and verify that the Twitter icon appears in the footer if the URL is provided.
3. Confirm that other social media icons are correctly displayed based on their respective environment variables.

### CODE REVIEW

**Comment by Indrajeety993648** [comment]
@rithviknishad   please review

**Review by rithviknishad — CHANGES_REQUESTED** [review]
Could you discard all changes that are unrelated to the related issue?

Feel free to create separate issues for each of the extra things solved.

**Comment by Indrajeety993648** [comment]
@rithviknishad  One prettier related problem  that  is necessary  to  solve during commit  that  I changed . Can u review and merge this  PR

**Comment by Indrajeety993648** [comment]
@rithviknishad  please review .

File: scraper/pnpm-lock.yaml
Comment (rithviknishad) [inline scraper/pnpm-lock.yaml:1]: Could you discard changes to the lockfile?

File: env.mjs [hunk 1]
```diff
@@ -56,10 +57,12 @@ export const env = createEnv({
         : process.env.GITHUB_PAT,
     NEXT_PUBLIC_FEATURES: process.env.NEXT_PUBLIC_FEATURES,
     NEXT_PUBLIC_DATA_SOURCE: process.env.NEXT_PUBLIC_DATA_SOURCE,
-    NEXT_PUBLIC_FLAT_REPO_EXPLORER_URL: process.env.NEXT_PUBLIC_FLAT_REPO_EXPLORER_URL,
+    NEXT_PUBLIC_FLAT_REPO_EXPLORER_URL:
+      process.env.NEXT_PUBLIC_FLAT_REPO_EXPLORER_URL,
     NEXT_PUBLIC_GITHUB_URL: process.env.NEXT_PUBLIC_GITHUB_URL,
     NEXT_PUBLIC_LINKEDIN_URL: process.env.NEXT_PUBLIC_LINKEDIN_URL,
     NEXT_PUBLIC_YOUTUBE_URL: process.env.NEXT_PUBLIC_YOUTUBE_URL,
+    NEXT_PUBLIC_TWITTER_URL: process.env.NEXT_PUBLIC_TWITTER_URL, // Add this line
```
Comment (rithviknishad) [inline env.mjs:65]:
```suggestion
    NEXT_PUBLIC_TWITTER_URL: process.env.NEXT_PUBLIC_TWITTER_URL,
```

File: env.mjs [hunk 2]
```diff
@@ -17,6 +17,7 @@ export const env = createEnv({
     NEXT_PUBLIC_GITHUB_URL: z.string(),
     NEXT_PUBLIC_LINKEDIN_URL: z.string(),
     NEXT_PUBLIC_YOUTUBE_URL: z.string(),
+    NEXT_PUBLIC_TWITTER_URL: z.string().optional(), // Add this line
```
Comment (rithviknishad) [inline env.mjs:20]:
```suggestion
    NEXT_PUBLIC_TWITTER_URL: z.string().optional(),
```

File: components/Footer.tsx [hunk 3]
```diff
@@ -94,26 +100,41 @@ export default function Footer() {
...
-                />
-                <SocialLink
-                  href={`mailto:${env.NEXT_PUBLIC_CONTACT_EMAIL}`}
-                  icon={FaEnvelope}
-                  label="Email"
-                />
+                {env.NEXT_PUBLIC_TWITTER_URL && (
+                  <SocialLink
+                    href={env.NEXT_PUBLIC_TWITTER_URL}
+                    icon={FaTwitter}
```
Comment (rithviknishad) [inline components/Footer.tsx:106]: You should be using the Twitter's X logo instead

**Review by rithviknishad — CHANGES_REQUESTED** [review]

**Pull request closed without merging**